src/squidpy/tl/_sliding_window.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+
+import math
+from functools import reduce
+from typing import TYPE_CHECKING, Any, Literal, NamedTuple, Optional, Union
+
+import numpy as np
+import pandas as pd
+from anndata import AnnData
+from scanpy import logging as logg
+
+from squidpy._docs import d
+from squidpy._utils import NDArrayA
+from squidpy.gr._utils import _save_data
+
+__all__ = ["sliding_window"]
+
+
+@d.dedent
+def sliding_window(
+    adata: AnnData,
```

Comment:
we're moving `squidpy` to be fully `spatialdata` compatible, so please add that here as well. Super simple, check: https://github.com/scverse/squidpy/blob/e6b5b0ccd2739aa50b55c7628e685a6750249d8f/src/squidpy/gr/_nhood.py#L159